examples/oauth.rs

```diff
@@ -0,0 +1,261 @@
+//! Example OAuth (Discord) implementation.
+//!
+//! Thanks to pbzweihander for their tide-google-oauth-example!
+//! It helped me understand oauth2.
+//!
+//! Run with
+//!
+//! ```not_rust
+//! CLIENT_ID=123123 CLIENT_SECRET=secret
+//! cargo run --example oauth --features=headers
```

Comment:
This being one line makes it easier to copy.

```suggestion
//! CLIENT_ID=123123 CLIENT_SECRET=secret cargo run --example oauth --features=headers
```